examples/kuka_iiwa_arm/iiwa_controller_demo.cc

```diff
@@ -25,68 +17,126 @@
+      std::ifstream stream;
+      stream.open(fileName);
+
+      //Checks to make sure it has been opened.
+      if (stream.is_open()) {
+        std::cout << "CSV file " + fileName + " opened successfully" << std::endl;
+      }
+
+      //Finds number of columns in CSV file to resize data vector.
+      std::string firstLine;
+      std::getline(stream, firstLine);
+      std::stringstream lineStream(firstLine);
+      std::string dat;
+      int length = 0;
+      while (std::getline(lineStream, dat, ',')) {
+        data.resize(data.size() + 1);
+        data[length].push_back(std::stod(dat));
+        length++;
+      }
+
+      //Loads CSV file contents into data vector.
+      while(std::getline(stream, firstLine)) {
+        int col = 0;
+        std::stringstream moreLines(firstLine);
+        while (std::getline(moreLines, dat, ',')) {
+          data[col].push_back(std::stod(dat));
+          col++;
+        }
+      }
+      stream.close();
+    }
+    //Returns the data array.
+    std::vector<std::vector<double>> getArray() {
+      return data;
+    }
+};
+
 // This function creates a controller for a Kuka LBR Iiwa arm by connecting an
 // EndEffectorPositionController to an EndEffectorVelocityController to control
 // the individual joint torques as to move the endeffector
 // to a desired position.
-int do_main(int argc, char* argv[]) {
-  // Creating end effector trajectory
-  // TODO make this modular
-  const std::vector<double> times {0.0, 25.0, 35.0, 45.0, 55.0, 65.0,
-                                   75.0, 85.0, 95.0, 105.0, 115};
-
-  std::vector<Eigen::MatrixXd> points(times.size());
-
-  Eigen::Vector3d AS, A0, A1, A2, A3, A4, A5, A6, A7, A8, A9;
-
-  AS << -0.23, -0.2, 0.25;
-  A0 << -0.23, -0.2, 0.25;
-
-  A1 << -0.23, -0.6, 0.25;
-  A2 << 0.23, -0.6, 0.25;
-
-  A3 << 0.23, -0.2, 0.25;
 
-  A4 << 0.23, -0.2, 0.25;
-
-  A5 << 0.23, -0.6, 0.25;
-
-  A6 << -0.23, -0.6, 0.25;
-
-  A7 << -0.23, -0.2, 0.25;
-
-  A8 << -0.23, -0.2, 0.25;
-  A9 << -0.23, -0.2, 0.25;
-
-  points[0] = AS;
-  points[1] = A0;
-  points[2] = A1;
-  points[3] = A2;
-  points[4] = A3;
-  points[5] = A4;
-  points[6] = A5;
-  points[7] = A6;
-  points[8] = A7;
-  points[9] = A8;
-  points[10] = A9;
-
-  auto ee_trajectory = drake::trajectories::PiecewisePolynomial<
-      double>::FirstOrderHold(times, points);
-
-  // Creating end effector orientation trajectory
-  const std::vector<double> orient_times {0, 115};
-  std::vector<Eigen::MatrixXd> orient_points(orient_times.size());
-  Eigen::Vector4d start_o, end_o;
-  start_o << 0, 0, 1, 0;
-  end_o << 0, 0, 1, 0;
-
-  orient_points[0] = start_o;
-  orient_points[1] = end_o;
-
-  auto orientation_trajectory = drake::trajectories::PiecewisePolynomial<
-      double>::FirstOrderHold(orient_times, orient_points);
+int do_main(int argc, char* argv[]) {
+  //Loads in joint gains json file
+  std::ifstream joint_gains_file("examples/kuka_iiwa_arm/simulationsettings.json");
```

Comment:
Not critical to change, but just some comments here.
- These file inputs could be command line arguments
- File directories can be sanitized using `dairlib::FindResourceOrThrow`. This would allow this code to be run from directories other than the main `dairlib/` location.
- Files in `drake/` can be found using `drake::FindResourceOrThrow`